keyboards/planck/keymaps/adamtabrams/keymap.c

```diff
@@ -208,7 +206,108 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
 };
 
+bool is_left_mod(uint16_t keycode) {
+    switch (keycode) {
+        case ALT__A:
+        case GUI__S:
+        case SFT__D:
+        case CTL__F:
+        case GUI__R:
+        case SFT__S:
+        case CTL__T:
+            return true;
+    }
+    return false;
+}
+
+bool is_right_mod(uint16_t keycode) {
+    switch (keycode) {
+        case ALTSCLN:
+        case GUI__L:
+        case SFT__K:
+        case CTL__J:
+        case ALT__O:
+        case GUI__I:
+        case SFT__E:
+        case CTL__N:
+            return true;
+    }
+    return false;
+}
+
+bool is_left_key(int row) {
+    return row < 4 || row == 7;
+}
+
+bool is_right_key(int row) {
+    return row > 2;
+}
+
+void tap_mods(int num_mods, uint16_t *mods, int *counts) {
+    clear_mods();
+    for (int i = 0; i < num_mods; i++) {
+        if (counts[i] == 0) {
+            tap_code16(mods[i]);
```

Comment:
This effectively does
```suggestion
            tap_code(QK_MOD_TAP_GET_TAP_KEYCODE(mods[i]));
```
However, that peculiarity of the `tap_code16()` behavior (keycodes outside of the `QK_BASIC`…`QK_MODS_MAX` range are truncated to the low byte) does not seem to be documented anywhere.